commands/team_users_e2e_test.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestTeamUserAddCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	user, appErr := s.th.App.CreateUser(&model.User{Email: s.th.GenerateTestEmail(), Username: model.NewId(), Password: model.NewId()})
+	s.Require().Nil(appErr)
+
+	team, appErr := s.th.App.CreateTeam(&model.Team{
+		DisplayName: "dn_" + model.NewId(),
+		Name:        api4.GenerateTestTeamName(),
+		Email:       s.th.GenerateTestEmail(),
+		Type:        model.TEAM_OPEN,
+	})
+	s.Require().Nil(appErr)
+
+	unlinkUserFromTeam := func(teamId string, userId string) error {
+		teamMembers, err := s.th.App.GetTeamMembers(teamId, 0, 10, nil)
+
+		if err != nil {
+			return err
+		}
+		var teamMember *model.TeamMember
+		for _, v := range teamMembers {
+			if v.UserId == userId {
+				teamMember = v
+				break
+			}
+		}
+		if teamMember == nil {
+			return nil
+		}
+		return s.th.App.RemoveTeamMemberFromTeam(teamMember, s.th.SystemAdminUser.Id)
+	}
+
+	s.RunForSystemAdminAndLocal("Add user to team", func(c client.Client) {
+		printer.Clean()
+
+		appErr := unlinkUserFromTeam(team.Id, user.Id)
+		s.Require().Nil(appErr)
+
+		err := teamUsersAddCmdF(c, &cobra.Command{}, []string{team.Id, user.Email})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+
+		teamMembers, err := s.th.App.GetTeamMembers(team.Id, 0, 10, nil)
+		s.Require().Nil(err)
+		s.Require().NotNil(teamMembers)
+		s.Require().Len(teamMembers, 1)
+		s.Require().Equal(user.Id, teamMembers[0].UserId)
+	})
+
+	s.Run("Add user to team without permissions", func() {
+		printer.Clean()
+
+		appErr := unlinkUserFromTeam(team.Id, user.Id)
+		s.Require().Nil(appErr)
+
+		err := teamUsersAddCmdF(s.th.Client, &cobra.Command{}, []string{team.Id, user.Email})
+		s.Require().NotNil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Require().Equal(fmt.Sprintf("Unable to find team '%s'", team.Id), err.Error())
+
+		teamMembers, err := s.th.App.GetTeamMembers(team.Id, 0, 10, nil)
+		s.Require().Nil(err)
+		s.Require().Len(teamMembers, 0)
+	})
+
+	s.Run("Add user to team with permissions", func() {
+		printer.Clean()
+
+		appErr := unlinkUserFromTeam(team.Id, user.Id)
+		s.Require().Nil(appErr)
+
+		_, appErr = s.th.App.AddTeamMember(team.Id, s.th.BasicUser.Id)
+		s.Require().Nil(appErr)
+
+		err := teamUsersAddCmdF(s.th.Client, &cobra.Command{}, []string{team.Id, user.Email})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+
+		teamMembers, err := s.th.App.GetTeamMembers(team.Id, 0, 10, nil)
+		s.Require().Nil(err)
+		s.Require().NotNil(teamMembers)
+		s.Require().Len(teamMembers, 2)
+
+		var teamUsersID []string
+		for _, v := range teamMembers {
+			teamUsersID = append(teamUsersID, v.UserId)
+		}
+		s.Require().Contains(teamUsersID, user.Id)
+
+		appErr = unlinkUserFromTeam(team.Id, s.th.BasicUser.Id)
```

Comment:
Should we defer this right after adding the member? Failing this tests can end up cascading in other failing tests because the database has this user.